File: backend/app/schemas/coach.py
from enum import Enum
from typing import Optional
from uuid import UUID

from app.schemas.base import CamelModel


class CoachContext(str, Enum):
    RESULTS_RECEIVED = "results_received"
    IDLE_NUDGE = "idle_nudge"
    HISTORY_VISIT = "history_visit"
    ON_DEMAND = "on_demand"
    GREETING = "greeting"


class CoachRequest(CamelModel):
    context: CoachContext
    submission_id: Optional[UUID] = None


class CoachResponse(CamelModel):
    message: str
    fox_state: str = "coaching"

File: src/api/coach.ts
import { apiClient } from "./client";

export type CoachContext =
  | "results_received"
  | "idle_nudge"
  | "history_visit"
  | "on_demand"
  | "greeting";

interface CoachResponse {
  message: string;
  foxState: string;
}

export async function getCoachingTip(
  context: CoachContext,
  submissionId?: string
): Promise<CoachResponse> {
  const { data } = await apiClient.post<CoachResponse>("/coach", {
    context,
    submissionId,
  });
  return data;
}
